packages/commonwealth/client/scripts/controllers/chain/ethereum/generalContracts.ts

```diff
@@ -0,0 +1,193 @@
+import $ from 'jquery';
+import m from 'mithril';
+
+import { Contract, NodeInfo, IWebWallet } from 'models';
+import { initAppState } from 'app';
+import { Contract as Web3Contract } from 'web3-eth-contract';
+import { parseAbiItemsFromABI, parseEventFromABI } from 'helpers/abi_utils';
+import { AbiItem } from 'web3-utils';
+import { ChainBase, ChainNetwork, ChainType } from 'common-common/src/types';
+import { linkExistingAddressToChainOrCommunity } from 'controllers/app/login';
+import {
+  ChainFormFields,
+  EthFormFields,
+} from 'views/pages/create_community/types';
+import { TransactionReceipt } from 'web3-core';
+import EthereumChain from './chain';
+
+type EthDaoFormFields = {
+  network: ChainNetwork.Ethereum;
+  tokenName: string;
+};
+type CreateFactoryEthDaoForm = ChainFormFields &
+  EthFormFields &
+  EthDaoFormFields;
+
+export default class GeneralContractsController {
```

Comment:
I actually think this is broadly the correct architecture, keeping the ContractsController separate from the ChainAdapter and injecting the API. This allows us to unit test by mocking the provided API object, for example.

However, I don't think it makes sense to inject the entire `EthereumChain`, since we only use the `.api` value + the global `app` object from it. So we should instead take a `Web3` object, ideally already initialized, and reference the global `app` (vs the value on chain.app or whatever) for the couple of cases it's needed.